File: typescript/secret-handshake/secret-handshake.ts
export function commands(secret:number) {
  let reverseActions = false
  const binary = secret.toString(2)
  const numbers = binary.padStart(5, "0").split('')
  const result: string[] = []
  const actions: string[] = [
    'undefined',
    'jump',
    'close your eyes',
    'double blink',
    'wink',
  ]
  
  for (let index = 0; index < numbers.length; index++) {
    if (index === 0 && numbers[0] === '1') {
      reverseActions = true
      continue
    }
    
    if (numbers[index] === '1') { 
      result.push(actions[index])
    }
  }

  if (reverseActions === true) {
    return result
  } else {
    return result.reverse()
  }
}

File: go/secret-handshake/secret_handshake.go
package secret

func Handshake(code uint) []string {
	results := []string{}

	if code&1 == 1 {
		results = append(results, "wink")
	}
	if code&2 == 2 {
		results = append(results, "double blink")
	}
	if code&4 == 4 {
		results = append(results, "close your eyes")
	}
	if code&8 == 8 {
		results = append(results, "jump")
	}
	if code&16 == 16 {
		for index, oppositeIndex := 0, len(results)-1; index < oppositeIndex; index, oppositeIndex = index+1, oppositeIndex-1 {
			results[index], results[oppositeIndex] = results[oppositeIndex], results[index]
		}
	}

	return results
}
